contracts/core/OriginCore.sol

```diff
@@ -36,25 +36,61 @@ contract OriginCore is OriginCoreInterface, OriginCoreConfig {
     /** The header of a meta-block. */
     struct Header {
 
+        Kernel kernel;
+        Transition transition;
+    }
+
+    /** The kernel of a meta-block header. */
+    struct Kernel {
+
         /** The height of this header's block in the chain. */
         uint256 height;
 
         /** The hash of this block's parent. */
         bytes32 parent;
 
+        /**
+         * The array of addresses of the validators that are updated within
+         * this block. Updated weights at the same index relate to the address
+         * in this array.
+         */
+        address[] _updatedValidators;
+
+        /**
+         * The array of weights that corresponds to the updated validators.
+         * Updated validators at the same index relate to the weight in this
+         * array. Weights of existing validators can only decrease.
+         */
+        uint256[] _updatedWeights;
+    }
+
+    /** The transition of a meta-block header. */
+    struct Transition {
+
+        /** A unique identifier that identifies what chain this vote is about. */
+        bytes20 _coreIdentifier;
```

Comment:
It is *sometimes* an address. At other times it could be a constant. For example, Ethereum is defined as core id `0x0000000000000000000000000000000000000001`, which is not an address.
So generally speaking the core id is not an address. But in some cases we use the address of a contract to specify the core id.